analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs

```diff
@@ -76,13 +79,24 @@ private static void CheckInvalidProperties(INamedTypeSymbol parameterType, Sonar
         var declaredProperties = new List<IPropertySymbol>();
         GetAllDeclaredProperties(parameterType, examinedTypes, declaredProperties);
         var invalidProperties = declaredProperties
-            .Where(x => !CanBeNull(x.Type) && !x.HasAttribute(KnownType.System_Text_Json_Serialization_JsonRequiredAttribute) && !x.IsRequired())
+            .Where(x => !IsExcluded(x))
             .Select(x => x.GetFirstSyntaxRef())
             .Where(x => !IsInitialized(x));
         foreach (var property in invalidProperties)
         {
             context.ReportIssue(Rule, property.GetIdentifier()?.GetLocation());
         }
+
+        static bool IsExcluded(IPropertySymbol property) =>
+            CanBeNull(property.Type)
+            || property.HasAnyAttribute(IgnoredAttributes)
+            || IsNewtonsoftJsonPropertyRequired(property)
+            || property.IsRequired();
+
+        static bool IsNewtonsoftJsonPropertyRequired(IPropertySymbol property) =>
+            property.GetAttributes(KnownType.Newtonsoft_Json_JsonPropertyAttribute).FirstOrDefault() is { } attribute
+            && attribute.TryGetAttributeValue("Required", out string valueType)
```

Comment:
This should work as well and is more efficent
```suggestion
            && attribute.TryGetAttributeValue("Required", out int required)
```

Reply:
What about https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonObjectAttribute_ItemRequired.htm on the model type.

Reply:
I would handle JsonObjectAttribute in another PR.
We should indeed support this as it can be applied to structs.
I'll test the behavior and open an issue.